### ISSUE TITLE
def miss_perry_s_compass(nodes)

### ISSUE THREAD

**Issue by Eurus-Holmes** [gumpu/TSP_Animation#2]
Hello @gumpu ,
In your anim.py , the function `def miss_perry_s_compass(nodes)` of the line 421, I did not understand how did it work...could you please explain it?

and following code:
```
# Is the node inside or outside the polygon?
        if point_in_poly(new_node.x, new_node.y, solution):
            idx_before = (index_in_list - 1) % i
            idx_after  = (index_in_list + 1) % i
            # it is Inside
            area1 = area_triangle(new_node, closest, solution[idx_before])
            area2 = area_triangle(new_node, closest, solution[idx_after])
            if area1 < area2:
                # Insert new node between closest and next
                pass
            else:
                # Insert 
                pass
            pass
        else:
            # it is outside
            pass
```

when `if area1 < area2:`, how to `Insert new node between closest and next`, the code is may incompelete...
How can I complete that? I went over the repo, but couldn't figure out how this might be possible. Your kind help will be highly appreciated.

Best regards,
Eurus-Holmes

**Comment by gumpu** [comment]
This is an algorithm suggested and invented by PerryF42.
There is a description of the algorithm [here](https://github.com/gumpu/TSP_Animation/commit/c507f6887578eeab857e7d0ed7d8f8cd907b00b9#commitcomment-16220885).
Tried to implement it, but there was a flaw in it.  Then there was an update to the algorithm, but that had a flaw too. Can't quite remember what it was though. Which is why I stopped with the algorithm.

Think the idea is a bit like putting a rubber band on the outside of all the cities and then pulling it in with many fingers to the center.


The second version of the algorithm was:

If it did not work, I have thought of an alternative:

Find coordinates of centre by finding the mean of all data points
Create 4 artificial data points, named A, B, C, and D, with coordinates A=(minimum x - 10%[maximum x - minimum x], minimum y - 10%[maximum y - minimum y]), B=(minimum x - 10%[maximum x - minimum x], maximum y + 10%[maximum y - minimum y]), C=(maximum x + 10%[maximum x - minimum x], minimum y - 10%[maximum y - minimum y]), D=(maximum x + 10%[maximum x - minimum x], maximum y + 10%[maximum y - minimum y]).
Order list of points by distance from centre, furthest first 
Plot data point 1
Plot data point 2
Connect with line
Plot data point 3
Connect with lines to data point 1 and data point 2
Plot data point 4
Connect with line to nearest data point. This nearest data point will already be connected to 2 other data points; connect data point 4 to both of these points; remove the longest line of the smallest triangle and the line which formed the second larger triangle.
Continue for all further data points as data point 4.
Remove the artificial data points A, B, C, and D, connecting the pairs of data points previously connected to each of the artificial data points.
